pkl-cli/pkl-cli.gradle.kts

```diff
@@ -210,8 +223,20 @@ fun Exec.configureExecutable(
         // executable
         if (!buildInfo.isReleaseBuild) {
           add("-Ob")
+        } else {
+          add("-Os")
+        }
+        if (buildInfo.isNativeArch) {
+          add("-march=native")
+        } else {
+          add("-march=compatibility")
+        }
+        if (buildInfo.isEnableOracleGraalvm) {
+          add("--gc=G1")
+          add("--enable-sbom=cyclonedx")
+        } else {
+          add("--gc=serial")
         }
-        add("-march=compatibility")
```

Comment:
changes here:

- when building a release...
  - `-Os` is passed, which optimizes for size (this is a new feature supported by `native-image`)
  - under oracle graalvm...
    - `--gc=G1 --enable-sbom=cyclonedx` are passed; the SBOM is embedded within the final binary
- when building with `-DnativeArch=true`...
  - `-march=native` is passed
  - otherwise, `-march=compatibility` is passed (no change)
- when building under graalvm CE...
  - `--gc=serial` is passed; this is the only feasible option aside from epsilon (no-op)
- when not building a release...
  - `-Ob` is passed to optimize build time (no change)